src/App.js

```diff
@@ -1,26 +1,144 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React from "react";
+import "./App.css";
+import TodoItems from "./todoitems";
+import Filters from "./filters";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      textLines: {
+        hello: false,
+        bye: false,
+        "another task": false,
+        "last task": false
+      },
+      completedItems: 0,
+      filter: null,
+      displayStyle: { display: "block" },
+      isSelect: ["all", "active", "completed"]
+    };
+  }
+
+  completingItems = () => {
+    this.setState(prevState => {
+      return { completedItems: prevState.completedItems + 1 };
+    });
+  };
+
+  leftIems = () => {
+    return Object.keys(this.state.textLines).length - this.state.completedItems;
```

Comment:
```suggestion
    return this.state.items.filter((item) => !item.completed).length;
```
Check it. Much easier when `items` are in array